src/components/UserProfile/Gen3Connection.js

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import { compose, withState } from 'recompose';
+
+import Button from 'uikit/Button';
+import step2Screenshot from 'assets/gen3TokenScreenshot.png';
+import { deleteSecret, setSecret } from 'services/secrets';
+import { GEN3 } from 'common/constants';
+import { getUser as getGen3User } from 'services/gen3';
+import { ModalFooter } from 'components/Modal/index.js';
+
+import { css } from 'emotion';
+import { injectState } from 'freactal';
+import XIcon from 'react-icons/lib/fa/close';
+import RightArrows from 'react-icons/lib/fa/angle-double-right';
+
+const modalWidth = 1000;
+const closeButtonWidth = 30;
+const styles = css`
+  span.numberBullet {
+    color: white;
+    background: #00afed;
+    width: 14px;
+    border-radius: 50%;
+    margin: 20px;
+    padding: 10px;
+    height: 14px;
+    display: inline-block;
+    text-align: center;
+  }
+
+  .tokenTitle {
+    padding: 20px;
+    color: #2b388f;
+    font-size: 18px;
+    font-weight: bold;
+  }
+`;
+
+const enhance = compose(
+  injectState,
+  withState('gen3Key', 'setGen3Key', ''),
+  withState('invalidToken', 'setInvalidToken', false),
+);
+
+const submitGen3Token = async ({ token, setIntegrationToken, onSuccess, onFail }) => {
+  //token = JSON.stringify(token);
+  token = token.replace(/\s+/g, '');
```

Comment:
Can you avoid mutation whereever possible? creating a new variable would be better